main/src/ca/uwaterloo/flix/language/phase/njvm/Mnemonics.scala

```diff
@@ -12,75 +15,133 @@ object Mnemonics {
 
   type **[R <: Stack, T] = StackCons[R, T]
 
-  class F[T] {
+  sealed trait JvmModifier {
+    def toInternal: Int = this match {
+      case JvmModifier.Public => ACC_PUBLIC
+      case JvmModifier.Private => ACC_PRIVATE
+      case JvmModifier.Protected => ACC_PROTECTED
+      case JvmModifier.Static => ACC_STATIC
+      case JvmModifier.Final => ACC_FINAL
+      case JvmModifier.Super => ACC_SUPER
+      case JvmModifier.Synchronized => ACC_SYNCHRONIZED
+      case JvmModifier.Volatile => ACC_VOLATILE
+      case JvmModifier.Bridge => ACC_BRIDGE
+      case JvmModifier.VarArgs => ACC_VARARGS
+      case JvmModifier.Transient => ACC_TRANSIENT
+      case JvmModifier.Native => ACC_NATIVE
+      case JvmModifier.Interface => ACC_INTERFACE
+      case JvmModifier.Abstract => ACC_ABSTRACT
+      case JvmModifier.Strict => ACC_STRICT
+      case JvmModifier.Synthetic => ACC_SYNTHETIC
+      case JvmModifier.Annotation => ACC_ANNOTATION
+      case JvmModifier.Enum => ACC_ENUM
+      case JvmModifier.Mandated => ACC_MANDATED
+      case JvmModifier.Deprecated => ACC_DEPRECATED
+
+      case JvmModifier.InvokeSpecial => INVOKESPECIAL
+    }
+  }
+  object JvmModifier {
+    case object Public extends JvmModifier
+    case object Private extends JvmModifier
+    case object Protected extends JvmModifier
+    case object Static extends JvmModifier
+    case object Final extends JvmModifier
+    case object Super extends JvmModifier
+    case object Synchronized extends JvmModifier
+    case object Volatile extends JvmModifier
+    case object Bridge extends JvmModifier
+    case object VarArgs extends JvmModifier
+    case object Transient extends JvmModifier
+    case object Native extends JvmModifier
+    case object Interface extends JvmModifier
+    case object Abstract extends JvmModifier
+    case object Strict extends JvmModifier
+    case object Synthetic extends JvmModifier
+    case object Annotation extends JvmModifier
+    case object Enum extends JvmModifier
+    case object Mandated extends JvmModifier
+    case object Deprecated extends JvmModifier
 
-    import org.objectweb.asm.MethodVisitor
-    import org.objectweb.asm.Opcodes
 
-    private val mv: MethodVisitor = ???
+    case object InvokeSpecial extends JvmModifier
+
+  }
 
-    def compile(): Array[Byte] = ???
 
-    def areturn[S](): F[S] = {
-      mv.visitInsn(Opcodes.ARETURN)
+  class F[T](mv : MethodVisitor, ct : JvmType.Reference) {
+
+    def load[S](localType: JvmType, location : Int): F[S] = {
```

Comment:
Is this some internal method or what? It should not be called directly on F right? Maybe it should be private or renamed to something like `__load`?